File: templates/java/src/main/java/com/adventofcode/Answer.java
package com.adventofcode;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

class Answer {

    public static String readInputFile() {
        StringBuilder strBuilder = new StringBuilder();
        int readVar;

        try (FileInputStream fileInput = new FileInputStream("input.txt")) {
            do {
                readVar = fileInput.read();
                if (readVar != -1) {
                    strBuilder.append((char) readVar);
                }
            } while (readVar != -1);
        } catch (IOException e) {
            System.out.println("I/O Error: " + e);
            return "";
        }

        return strBuilder.toString();
    }

    public static List<Integer> splitAndConvertToIntegers(String data) {
        return Arrays.stream(data.split("\n")).map(Integer::valueOf).collect(Collectors.toList());
    }

    public int solution() {
        String[] data = readInputFile().split("\n");
        int result = 0;

        // CODE

        return result;
    }

    public static void main(String[] args) {
        System.out.println(new Answer().solution());
    }

}

File: day-08-1/ruby/answer.rb
# frozen_string_literal: true

class Answer
  def self.read_input_file
    File.read('input.txt')
  end

  def main
    entries = self.class.read_input_file.split("\n").map do |entry|
      entry.split(' | ')
    end

    four_digit_output_values = entries.map(&:last).map(&:split)

    four_digit_output_values.flatten.select do |value|
      [2, 3, 4, 7].include? value.size
    end.size
  end
end

puts Answer.new.main
